src/Actions/Contexts/ScopedActionContext.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Axe.Windows.Desktop.UIAutomation.CustomObjects;
+using Axe.Windows.Desktop.UIAutomation.TreeWalkers;
+using System;
+using System.Threading;
+
+namespace Axe.Windows.Actions.Contexts
+{
+    /// <summary>
+    /// An implementation of <see cref="IActionContext"></see> for scoped duration.
+    /// Object cleanup occurs when the object is Disposed.
+    /// </summary>
+    internal class ScopedActionContext : IActionContext
+    {
+        private bool disposedValue;
+
+        private ScopedActionContext(DataManager dataManager, SelectAction selectAction, TreeWalkerDataContext treeWalkerDataContext)
+        {
+            DataManager = dataManager ?? throw new ArgumentNullException(nameof(dataManager));
+            SelectAction = selectAction ?? throw new ArgumentNullException(nameof(selectAction));
+            TreeWalkerDataContext = treeWalkerDataContext ?? throw new ArgumentNullException(nameof(treeWalkerDataContext));
+        }
+
+        public DataManager DataManager { get; }
+
+        public SelectAction SelectAction { get; }
+
+        public Registrar Registrar => TreeWalkerDataContext.Registrar;
+
+        public TreeWalkerDataContext TreeWalkerDataContext { get; }
+
+        protected virtual void Dispose(bool disposing)
```

Comment:
General question: What is the purpose of the `bool disposing` parameter, here and throughout the code base? In what instance is it false?

Reply:
Most of this is VS boilerplate. VS had created this code blurb that I deleted but have since restored:

```
        // TODO: override a finalizer only if Dispose(bool disposing) above has code to free unmanaged resources.
        // ~ScopedActionContext() {
        //   // Do not change this code. Put cleanup code in Dispose(bool disposing) above.
        //   Dispose(false);
        // }
```

If this code blurb is uncommented, then the `disposing` parameter would be false if the object were garbage collected before its `Dispose` method was called. A derived class can also call the `Dispose` method with `disposing` = false. In this particular class, neither of those cases exist, so we could reasonably remove the parameter. Most of the time, though, we just leave the default boilerplate as it was originally created. The boilerplate covers the broadest general case, even if it's generally not fully leveraged. There's a fair write-up of this pattern at https://vkontech.com/the-dispose-pattern-step-by-step/, and I'm happy to chat about it if you have other questions 😃